fiatconnect-api.md

```diff
@@ -1710,14 +1713,78 @@ This endpoint MUST fail when the user has no transfer on file with the provided
 
 If the user has no transfer on file with the provided `transferId`, the server MUST return a `ResourceNotFound` error.
 
-# 4. Webhooks
+# 4. Transfer State Machines
+
+The FiatConnect specification requires that both transfers in and out progress through a particular series of states that are made
+available to the client both by polling the `GET /transfer/:transferId/status` endpoint and via webhooks. Statuses for transfers in and
+transfers out must progress according to a specific *state machine*. This state machine is different for both transfers in and transfers out.
+
+## 4.1. Transfers In
+
+Statuses for transfers in must progress through the state machine shown below, ultimately ending at one of the two *terminal* states:
+`TransferFailed` or `TransferComplete`. A server SHOULD make a best effort to reach a terminal transfer state within the `settlementTimeUpperBound`
+specified for the quote associated with the transfer, if one exists. Note that since a transfer in may fail after the server has received fiat funds
+from the user's fiat account but before crypto funds are sent to the user, the server MUST return any fiat funds to the user's fiat account in case
+of such a failure.
+
+```mermaid
+graph LR
+   TransferStarted---->TransferPending
+   TransferPending---->TransferFiatFundsDebited
+   TransferFiatFundsDebited---->TransferWaitingForFiatFunds
+   TransferWaitingForFiatFunds---->TransferReceivedFiatFunds
+   TransferReceivedFiatFunds---->TransferSendingCryptoFunds
```

Comment:
as a provider, would I ever want to do anything between `TransferReceivedFiatFunds` and `TransferSendingCryptoFunds`? (Could they be one state?)

Reply:
Yeah, they could probably safely be merged... I wanted to have clear events for receipt of fiat funds and transfer of crypto funds, but realistically, these two things will happen back to back.